### DISCUSSION TITLE
Simple example with another language

### DISCUSSION THREAD

**Discussion by alex-hedley** [microsoft/monaco-editor#3872] [Q&A] [answered]
How do I work with other languages?
Looking at the [samples](https://github.com/microsoft/monaco-editor/tree/main/samples) they all use `javascript`.

I tried the [browser-amd-editor](https://github.com/microsoft/monaco-editor/blob/main/samples/browser-amd-editor/index.html) example.

Is it enough to just update

```js
var editor = monaco.editor.create(document.getElementById('container'), {
  value: ['Console.WriteLine("Hello, World!");'].join('\n'),
  language: 'csharp'
});
```

As when I type in code there isn't the autocomplete showing.

**Comment by alex-hedley** [comment]
I tried the full C# example from https://microsoft.github.io/monaco-editor/ which contained the `namespace` etc and then IntelliSense worked.

Does it not support Top Level Statements?

**Comment by tamayika** [comment, marked as answer]
monaco-editor has language server protocol support only for `json`, `html`, `css`, `typescript`, `javascript`.
You can see keyword completions as you ever typed before in editor for other languages.
See: https://github.com/microsoft/monaco-editor/tree/main/src/language
